src/ZkFold/Symbolic/Data/Ed25519.hs

```diff
@@ -1,115 +1,108 @@
+{-# LANGUAGE DerivingStrategies   #-}
+{-# LANGUAGE DerivingVia          #-}
 {-# LANGUAGE TypeOperators        #-}
 {-# LANGUAGE UndecidableInstances #-}
 
 {-# OPTIONS_GHC -Wno-orphans #-}
 
 module ZkFold.Symbolic.Data.Ed25519  where
 
-import           Data.List                                 (splitAt)
 import           Data.Void                                 (Void)
+import           GHC.Generics                              (Generic1, Generically1 (..))
 import           GHC.TypeNats                              (Natural)
-import           Prelude                                   (error, otherwise, ($), (.), (<>))
+import           Prelude                                   (($), (.), (<>))
 import qualified Prelude                                   as P
 
 import           ZkFold.Base.Algebra.Basic.Class
 import           ZkFold.Base.Algebra.Basic.Field
 import           ZkFold.Base.Algebra.Basic.Number
-import           ZkFold.Base.Algebra.EllipticCurve.Class
-import           ZkFold.Base.Algebra.EllipticCurve.Ed25519
-import           ZkFold.Prelude                            (length)
+import           ZkFold.Base.Algebra.Basic.VectorSpace
+import qualified ZkFold.Base.Algebra.EllipticCurve.Class   as Base
+import qualified ZkFold.Base.Algebra.EllipticCurve.Ed25519 as Base
 import           ZkFold.Symbolic.Compiler                  hiding (forceZero)
 import           ZkFold.Symbolic.Data.Bool
 import           ZkFold.Symbolic.Data.Combinators
-import           ZkFold.Symbolic.Data.Conditional
-import           ZkFold.Symbolic.Data.DiscreteField
-import           ZkFold.Symbolic.Data.Eq
+import           ZkFold.Symbolic.Data.Compare
 import           ZkFold.Symbolic.Data.UInt
 
-zpToEd :: (Finite (Zp p)) => Point (Ed25519 (Zp p)) -> Point (Ed25519 Void)
-zpToEd Inf         = Inf
-zpToEd (Point x y) = Point (toZp . toConstant $ x) (toZp . toConstant $ y)
-
-edToZp :: (Finite (Zp p)) => Point (Ed25519 Void) -> Point (Ed25519 (Zp p))
-edToZp Inf         = Inf
-edToZp (Point x y) = Point (fromConstant . fromZp $ x) (fromConstant . fromZp $ y)
+data PointEd25519 a = PointEd25519 (UInt 256 a) (UInt 256 a)
+  deriving stock (P.Eq, P.Functor, P.Foldable, P.Traversable, Generic1)
+deriving via Generically1 PointEd25519 instance FiniteField a => VectorSpace a PointEd25519
+instance (DiscreteField a, FiniteField a) => Eq a PointEd25519
+instance (TrichotomyField a, FiniteField a) => Ord a PointEd25519
```

Comment:
Wait, how does Ord work for two-dimensional points? Lexicografically?
Anyway, do we need this instance?

Reply:
Yes, lexicographically. Perhaps we don't need it.